src/Cake.Common/Tools/DotNetCore/DotNetCoreResolver.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System.Diagnostics;
+using Cake.Core;
+using Cake.Core.IO;
+
+namespace Cake.Common.Tools.DotNetCore
+{
+    internal static class DotNetCoreResolver
+    {
+        private static ICakeEnvironment _environment;
+
+        public static FilePath GetDotNetCorePath(ICakeEnvironment environment)
+        {
+            _environment = environment;
+
+            if (_environment.IsUnix())
+            {
+                return GetWhichDotNet();
+            }
+            else
+            {
+                return GetWhereDotNet();
+            }
+        }
+
+        private static FilePath GetWhichDotNet()
+        {
+            var startInfo = new ProcessStartInfo
+            {
+                FileName = "/usr/bin/which",
+                Arguments = "dotnet",
+                UseShellExecute = false,
+                RedirectStandardOutput = true,
+                CreateNoWindow = true
+            };
+
+            string which;
+            using (var process = new Process { StartInfo = startInfo })
+            {
+                process.Start();
+                which = process.StandardOutput.ReadToEnd();
+            }
+
+            return string.IsNullOrEmpty(which) ? null : new FilePath(which.Trim());
+        }
+
+        private static FilePath GetWhereDotNet()
```

Comment:
On Windows it was. Had some troubles on my Mac, but that was a couple of pre-releases ago. Let me retest and change it.